pkg/sql/catalog/tabledesc/table_desc_builder.go

```diff
@@ -671,13 +671,6 @@ func maybeRemoveDuplicateIDsInRefs(d *descpb.TableDescriptor) (hasChanged bool)
 	for i := range d.DependedOnBy {
 		ref := &d.DependedOnBy[i]
 		s := catalog.MakeTableColSet(ref.ColumnIDs...).Ordered()
-		// Also strip away O-IDs, which may have made their way in here in the past.
```

Comment:
Just wanted to make sure I understood it correctly -- the purpose of keeping the 0 column id is that the user can be aware that there are dependents that can't be back referenced, and thus they will do manual changes. Is this correct?

Reply:
This was some postDeserializationChanges code we used to have where we want to duduplicate column IDs. We used to have some additional logic that says "if there are any 0-valued column ID and non-zero valued column IDs, we will remove the 0-valued column ID". I'm not sure why we had that but that had caused some of my newly added test to fail (where we would construct a corrupt sequence descriptor with a back reference with column ID [2, 0]) so I removed it.

Reply:
when was this code added? Does this mean we need to check on back-references when there are no referencing column IDs?